lerobot/common/datasets/utils.py

```diff
@@ -230,6 +232,56 @@ def load_previous_and_future_frames(
     return item
 
 
+def calculate_episode_data_index(hf_dataset: datasets.Dataset) -> Dict[str, torch.Tensor]:
+    """
+    Calculate episode data index for the provided HuggingFace Dataset. Relies on episode_index column of hf_dataset.
+
+    Parameters:
+    - hf_dataset (datasets.Dataset): A HuggingFace dataset containing the episode index.
+
+    Returns:
+    - episode_data_index: A dictionary containing the data index for each episode. The dictionary has two keys:
+        - "from": A tensor containing the starting index of each episode.
+        - "to": A tensor containing the ending index of each episode.
+    """
+    episode_data_index = {"from": [], "to": []}
+
+    current_episode = None
+    episode_index = hf_dataset['episode_index']
+    for idx, num_episode in enumerate(episode_index):
+        if num_episode != current_episode:
+            episode_data_index["from"].append(idx)
+            if current_episode is not None:
+                episode_data_index["to"].append(idx)
+            current_episode = num_episode
+        elif idx == len(episode_index) - 1:
+            episode_data_index["to"].append(idx + 1)
```

Comment:
We could remove `episode_index` since it's only used once

```suggestion
    for i, num_episode in enumerate(hf_dataset['episode_index']):
```

Also I am a bit confused in term of naming. We usually use `num_*` to refer to a length of possible list. Here `num_episode` refers to an index.

Could we find a way to make this code logic more readable?